### PR TITLE
fix: function paramter list can end in comma. Fixes# 613

### LINKED ISSUE #613: [BUG] Parameter list can end in comma
<a id="title"/>

**[T](#title)itle**: Parameter list can end in comma.

<a id="rep"/>
<a id="cpp2"/>

**[M](#rep)inimal reproducer** (<https://cpp2.godbolt.org/z/oz6c57P8n>): `main: (args,) = { }`

<a id="cmd"/>
<details><summary>
<b><a href="#cmd">C</a>ommands</b>:
</summary>

```bash
cppfront main.cpp2
clang++17 -std=c++23 -stdlib=libc++ -lc++abi -pedantic-errors -Wall -Wextra -Wconversion -Werror=unused-result -I . main.cpp
```
</details>

<a id="expected"/>

**[E](#expected)xpected result**: Similar to `main: () = _ = :i32 = (0,);`: `error: invalid text in expression list (at ')')`.

<a id="actual"/>

**[A](#actual)ctual result and error**: `Program returned: 0`.

### PR DESCRIPTION
**Pull request by jobnz**
Fixes #613 by checking if parameter-declaration list ended in a comma

### CODE REVIEW

**Comment by hsutter** [comment]
Thanks for your pull request! It looks like this may be your first contribution to cppfront. I've sent the Contributor License Agreement (CLA) to your email, and once it's signed I can look at your pull request. Thanks again for your contribution.

File: regression-tests/pure2-bugfix-for-parameter-decl-list.cpp2 [hunk 1]
```diff
@@ -0,0 +1 @@
+main: (args,) = {}
```
Comment (JohelEGP) [inline regression-tests/pure2-bugfix-for-parameter-decl-list.cpp2:1]: The test name should end in `-error.cpp2`, like the error Cpp2-time error tests.

Comment (jobnz) [reply regression-tests/pure2-bugfix-for-parameter-decl-list.cpp2]: Oh, right. Thanks!

**Comment by hsutter** [comment]
Looks good, thanks!

**Pull request merged**